### PR TITLE
Fix possibly disappearing airwires

### PR DESCRIPTION
**Pull request by ubruhin**
It's actually not a fix, but a workaround for issue #588. The Delaunay triangulation library is not able to handle colinear points in the input data (our net points) and silently ignores some of the input points, so there are edges (our airwires) missing in the output.

I spend quite some time on this issue, and I think it's not possible to actually fix this problem on our side, so the fix needs to be done in [delaunay-triangulation](https://github.com/Bl4ckb0ne/delaunay-triangulation). But actually I'm not sure if it's worth investigating and fixing the issue in that library - in my opinion the overall quality of that library is not acceptable for an EDA tool anyway. For long-term, we might switch to a different library, for example [delaunator-cpp](https://github.com/abellgithub/delaunator-cpp) which seems to be developed more seriously. But they currently also have a similar issue (https://github.com/abellgithub/delaunator-cpp/issues/8), so we should wait until this issue is resolved.

However, to get LibrePCB 0.1.4 released as soon as possible with disappearing airwires fixed, I simply implemented some kind of "fallback". Even if delaunay-triangulation failed to add some edges, we now still get all edges connected, but connections might not be "optimal", i.e. not the shortest distance is shown. But at least no airwires disappear now, so the risk of producing PCBs with missing connections is gone.

I also refactored the airwires builder to make it easier testable, and added some unittests which are able to detect the current bug.

### CODE REVIEW

File: libs/librepcb/common/algorithm/airwiresbuilder.cpp [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+              return a.weight > b.weight;
+            });
+
+  while (mstSize < mstExpectedSize && !mEdges.empty()) {
+    auto& dt = mEdges.back();
+
+    int srcTag = tags[dt.p1.id];
+    int trgTag = tags[dt.p2.id];
+    // printf("mstSize %d %d %f %d<>%d\n", mstSize, mstExpectedSize,
+    // dt.weight, srcTag, trgTag);
```
Comment (dbrgn) [inline libs/librepcb/common/algorithm/airwiresbuilder.cpp:135]: Driveby comment (I didn't really look at the logic): There's quite some commented out debug code in here (more below). Do you want to keep that?

Comment (ubruhin) [reply libs/librepcb/common/algorithm/airwiresbuilder.cpp:135]: Yeah I don't like such comments too, but as stated above the function, this function is copied from Horizon (which copied it from KiCad) and I'd just like to keep it as-is to keep the diff to the upstream code as small as possible :wink:

Comment (dbrgn) [reply libs/librepcb/common/algorithm/airwiresbuilder.cpp:135]: Ah, good point, I missed that! :+1:
